client/asset/eth/multirpc.go

```diff
@@ -502,6 +508,34 @@ func (m *multiRPCClient) connect(ctx context.Context) (err error) {
 	return nil
 }
 
+// goodNonce returns true when a nonce has not been received recently.
+func (m *multiRPCClient) goodNonce(nonce uint64) bool {
```

Comment:
It's really tempting just to forego the server and handle our own nonces, at least after some initial nonce resolution step of some kind. I wonder how bad it would be.

Reply:
Or a combination of both. I think the problem there is if there is an error sending the tx the nonce would not go up and we might not be able to know that all the time, without asking the chain. The cache I added here has a minute timeout for that reason. Maybe an error happened and the nonce we thought we used never really got mined.